winpr/libwinpr/utils/image.c

```diff
@@ -169,10 +178,16 @@ int winpr_bitmap_write(const char* filename, const BYTE* data, size_t width, siz
 	if (!bmp_header)
 		goto fail;
 
-	if (fwrite(bmp_header, WINPR_IMAGE_BMP_HEADER_LEN, 1, fp) != 1 ||
-	    fwrite((const void*)data, img_size, 1, fp) != 1)
+	if (fwrite(bmp_header, WINPR_IMAGE_BMP_HEADER_LEN, 1, fp) != 1)
 		goto fail;
 
+	for (size_t y = 0; y < height; y++)
+	{
+		const void* line = &data[stride * y];
+		if (fwrite(line, bpp_stride, 1, fp) != 1)
```

Comment:
Should be `width` as we are not saving stride anywhere in the bmp and we will load garbage when reading the file